### PR TITLE
Add Session scripts

### PR DESCRIPTION
**Pull request by jamesjlyons**
## Description

Add commands to control Session app

## Type of change

- [x] New script command

## Screenshot

Cleanshot gif demo:
https://cln.sh/lkxhCO

## Dependencies / Requirements

Requires Session 2.1.6 https://www.stayinsession.com/

## Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: commands/apps/session/session-start-new.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Optional parameters:
+# @raycast.icon images/session.png
+# @raycast.packageName Session
+
+# @Documentation:
+# @raycast.description Starts a focus session in Session app
+# @raycast.author James Lyons
+# @raycast.authorURL https://github.com/jamesjlyons
+
+# @raycast.argument1 { "type": "text", "placeholder": "Intent", "optional": true }
```
Comment (unnamedd) [inline commands/apps/session/session-start-new.sh:19]: This is something you don't need to use: `// /%20`. It is solved internally by Raycast just using a property here:
```suggestion
# @raycast.argument1 { "type": "text", "placeholder": "Intent", "optional": true, "percentEncoded": true }
```

File: commands/apps/session/session-start-new.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# @raycast.argument2 { "type": "text", "placeholder": "Duration", "optional": true }
+
+open "session:///start?intent=${1// /%20}&duration=${2}"
```
Comment (unnamedd) [inline commands/apps/session/session-start-new.sh:22]:
```suggestion
open "session:///start?intent=${1}&duration=${2}"
```

File: commands/apps/session/session-start-new.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# @raycast.icon images/session.png
+# @raycast.packageName Session
+
+# @Documentation:
+# @raycast.description Starts a focus session in Session app
+# @raycast.author James Lyons
+# @raycast.authorURL https://github.com/jamesjlyons
+
+# @raycast.argument1 { "type": "text", "placeholder": "Intent", "optional": true }
+# @raycast.argument2 { "type": "text", "placeholder": "Duration", "optional": true }
```
Comment (unnamedd) [inline commands/apps/session/session-start-new.sh:20]: I'm not sure, but I guess the duration in Session accept minutes, right? I guess an indication here might be good.

```suggestion
# @raycast.argument2 { "type": "text", "placeholder": "Minutes duration", "optional": true }
```

File: commands/README.md [hunk 4]
```diff
@@ -133,6 +134,16 @@ This repository contains sample commands and documentation to write your own one
...
+  | <img src="https://raw.githubusercontent.com/raycast/script-commands/master/commands/apps/session/images/session.png" width="20" height="20"> | [Pause Session](apps/session/session-pause.sh) | Starts a focus session in Session app | [James Lyons](https://github.com/jamesjlyons) |  |  | <img src="images/icon-bash.png" width="20" height="20" title="Bash"> |
+  | <img src="https://raw.githubusercontent.com/raycast/script-commands/master/commands/apps/session/images/session.png" width="20" height="20"> | [Start Previous Session](apps/session/session-start-previous.sh) | Starts a focus session in Session app with the previous intent and duration | [James Lyons](https://github.com/jamesjlyons) |  |  | <img src="images/icon-bash.png" width="20" height="20" title="Bash"> |
+  | <img src="https://raw.githubusercontent.com/raycast/script-commands/master/commands/apps/session/images/session.png" width="20" height="20"> | [Start Session](apps/session/session-start-new.sh) | Starts a focus session in Session app | [James Lyons](https://github.com/jamesjlyons) | ✅ |  | <img src="images/icon-bash.png" width="20" height="20" title="Bash"> |
+
 #### Sip
```
Comment (unnamedd) [inline commands/README.md:147]: This file is auto-generated. You don't need to change it.

File: commands/apps/session/session-abandon.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Note: Session X v2.1.6 required: https://www.stayinsession.com/
```
Comment (unnamedd) [inline commands/apps/session/session-abandon.sh:3]: To keep the conformance with our current style to indicate that your Script Command has an external dependency, let change the above line and in the other files to the suggestion:

```suggestion
# Dependency: This script requires `Session X v2.1.6` installed: https://www.stayinsession.com/
```

File: commands/apps/session/session-abandon.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# Optional parameters:
+# @raycast.icon images/session.png
+# @raycast.packageName Session
+
+# @Documentation:
+# @raycast.description Starts a focus session in Session app
+# @raycast.author James Lyons
+# @raycast.authorURL https://github.com/jamesjlyons
+
+open "session:///abandon"
```
Comment (unnamedd) [inline commands/apps/session/session-abandon.sh:19]: This is the first time I see three slash bars in the URL Scheme. Question: is this working nicely? I've tried here, but this is a Pro feature and I could test if it is working properly or not.

File: commands/extensions.json [hunk 7]
```diff
@@ -6574,6 +6574,132 @@
...
+              "isTemplate" : false,
+              "language" : "bash",
+              "mode" : "silent",
+              "needsConfirmation" : null,
+              "packageName" : "Session",
+              "refreshTime" : null,
+              "schemaVersion" : 1,
+              "title" : "Abandon Session"
+            }
+          ]
```
Comment (unnamedd) [inline commands/extensions.json:6702]: This file is auto-generated, you don't need to modify it.

**Review by unnamedd — CHANGES_REQUESTED** [review]
Hey @jamesjlyons, nice addition, even if I couldn't test in my Session version (I'm not Pro user).

Comments made, check it out when possible. If you need anything, please, let us know.

File: commands/apps/session/session-start-new.sh (hunk 3 above)
Comment (jamesjlyons) [reply commands/apps/session/session-start-new.sh:20]: You would be right, good call!

File: commands/README.md (hunk 4 above)
Comment (jamesjlyons) [reply commands/README.md:147]: I didn't manually change this file, it seems like it was generated when I initiated the pull request. Same with the 'commands/extensions.json' file. I received this email, if it's of any help:
CleanShot 2021-03-18 at 09 44 25@2x

File: commands/apps/session/session-abandon.sh (hunk 6 above)
Comment (jamesjlyons) [reply commands/apps/session/session-abandon.sh:19]: Same, but it's working fine here! I pulled it from Session's docs.

File: commands/extensions.json (hunk 7 above)
Comment (jamesjlyons) [reply commands/extensions.json:6702]: See comment above on `commands/README.md`

File: commands/README.md (hunk 4 above)
Comment (unnamedd) [reply commands/README.md:147]: Ahhh okay, I know what happened. The GitHub Actions runs in the `master` branch and your implementations was done in this branch, then the action was activated! If you have worked in a separated branch, this wouldn't happen. 

Thanks a lot for let me know of that!

File: commands/apps/session/session-abandon.sh (hunk 6 above)
Comment (unnamedd) [reply commands/apps/session/session-abandon.sh:19]: Great! Thanks for the answer! :D

File: commands/README.md (hunk 4 above)
Comment (jamesjlyons) [reply commands/README.md:147]: Got it! I'll go back to working from branches in the future. Do you need me to start a new pull request or can we ignore these files on merge?

Comment (unnamedd) [reply commands/README.md:147]: Don't worry @jamesjlyons, it will be overwritten anyway!

**Comment by unnamedd** [comment]
@jamesjlyons as we have conflicts with `extensions.json` you will have to merge merge the master branch from this repo to yours to solve the conflict before the Raycast team be able to merge your pull request here. You don't need to close this pull request and create another, just try to fix the problem and then they should merge soon.

**Comment by jamesjlyons** [comment]
@unnamedd Alright, conflicts should be fixed. Thanks for the help 🙌

**Comment by fe9lix** [comment]
Any updates here?

**Comment by jamesjlyons** [comment]
> Any updates here?

I had the conflicts resolved but it looks like there's more now. I'll merge the main branch in again

**Comment by jamesjlyons** [comment]
@fe9lix should be good to merge again

**Review by fe9lix — APPROVED** [review]

**Pull request merged**